misc/systemd/ipfs.service

```diff
@@ -3,9 +3,49 @@ Description=InterPlanetary File System (IPFS) daemon
 After=network.target
 
 [Service]
```

Comment:
Let's make the default security features the minimal subset that don't break go-ipfs features. Then we can have additional commented out sections: "if you don't need feature X, you can enable ...".

Reply:
I'm split on this. I think hardening is crucial on network services while the mount feature isn't used that often.

I wonder if it's not the better solution to slice the mount feature out of go-ipfs and have a separate daemon just querying the local API and mount the filesystem.

Since fuse is quite a deep integration in the kernel, there's quite a security risk to run a daemon which also talks to the web IMHO.

Until than, how about adding a flag which enables the fuse functionality, or disables it, like `--manage-fdlimit` can disable the changes to the fd-limit.

---

That said, the `ipfs` system user isn't in the fuse group, so manual intervention is necessary anyway to use the fuse feature, regardless of the security settings.

Reply:
> I wonder if it's not the better solution to slice the mount feature out of go-ipfs and have a separate daemon just querying the local API and mount the filesystem.

It is, but that's not how it works now.

I agree that hardening is important, but I'm not going to ship a default unit that doesn't work with some go-ipfs features. We can always add a commented out section explaining how to enable the additional hardening features.

> That said, the ipfs system user isn't in the fuse group, so manual intervention is necessary anyway to use the fuse feature, regardless of the security settings.

Hm. That we should fix by adding `fuse` to the supplementary groups. Manual intervention is still required to allow accessing the mount points by other users, but that's easily discoverable.

Reply:
> That we should fix by adding `fuse` to the supplementary groups

Right, but that's out of the scope of this PR. I'll open a new one.